formula/src/main/java/com/instacart/formula/internal/Listeners.kt

```diff
@@ -40,4 +38,30 @@ internal class Listeners {
         }
         listeners?.clear()
     }
+
+    /**
+     * Function which returns next index for a given key. It will
+     * mutate the [indexes] map.
+     */
+    private fun nextIndex(key: Any): Int {
+        val indexes = indexes ?: run {
+            val initialized = mutableMapOf<Any, Int>()
+            this.indexes = initialized
+            initialized
+        }
```

Comment:
Maybe I'm missing something, but looks like this could be simplified to `val indexes = this.indexes ?: mutableMapOf<Any, Int>()`?

Reply:
We need to assign `indexes` variable after we initialize the map 
```kotlin
this.indexes = initialized
```

Reply:
We are using lazy instantiation to avoid unnecessary allocations internally.